server/src/test/java/org/elasticsearch/usage/UsageServiceTests.java

```diff
@@ -41,6 +42,45 @@
 
 public class UsageServiceTests extends ESTestCase {
 
+    public void testNullHandler() {
```

Comment:
How do you feel about adding JavaDoc for these new test cases? We added something to the contribution guidelines about this, and I personally find it enormously useful to check that the stated purpose of the test matches the implementation.

Reply:
Also, and maybe this is just a preference, but I like to name my tests so that they contain the expectation - so `testHandlerCannotBeNull`, or `testHandlerMustHaveAName`, etc. Otherwise the test name says what the test is doing, but not what the outcome should be. There's overlap with the JavaDoc, but it might be useful in e.g. the Gradle UI.

Reply:
I added Javadocs, and I adjusted the name of `testHandlerCannotBeNull`, but `testAHandlerWithNoName` I'm keeping because it's a reference to a song, and it's fun.